### ISSUE TITLE
[gtk] Update to 4.0

### ISSUE THREAD

**Issue by pquiring** [microsoft/vcpkg#7212]
gtk4 isn't complete yet but adding it would help port existing gtk3 code to gtk4.

Thanks,

**Comment by handicraftsman** [comment]
Yes, that would be pretty nice.

**Comment by WorkingRobot** [comment]
gtk4 has been officially released yesterday. Seems like a good idea to look at it now

**Comment by faserg1** [comment]
I'll try to port this one, but for first port I'll need some help here.

**Comment by WorkingRobot** [comment]
Due to an [issue](https://gitlab.gnome.org/GNOME/gtk/-/issues/3480) with GTK4 and Win32, I think it may be a good idea to hold off for a little bit before starting.

**Comment by handicraftsman** [comment]
> Due to an [issue](https://gitlab.gnome.org/GNOME/gtk/-/issues/3480) with GTK4 and Win32, I think it may be a good idea to hold off for a little bit before starting.

Win32 isn't the only platform Gtk and vcpkg target. Gtk is mostly used (and has full feature support) on Linux desktops, while vcpkg definitely supports this OS.
And while gtk3 can be found in every single distro at the moment, gtk4 seems to be only available in fedora and probay arch. Ubuntu 20.04 will probably never include it.
Flatpaks and snaps allow gtk4 apps on any distro without depending on system gtk, but much people prefer to not use them.

**Comment by handicraftsman** [comment]
You could just block installing it on win32 until the bug is fixed.

**Comment by japm48** [comment]
Hopefully will be fixed by #13100.

**Comment by NancyLi1013** [comment]
Since we have PR #13100 to update `gtk `to 4.0.1. There is no need to add a new port `gtk4`. 
Another issue #11797 also track the updates for `gtk`. Let's move attention there.

**Comment by vserdyuk** [comment]
Updating existing gtk port to version 4 was an inconvenience for us since there's a lot of API breakage. Here's overrides we use in manifest for keeping gtkmm to be the last 3.x version (in case someone finds it useful):

```json
  "overrides":  [
    { "name": "atk", "version-string": "2.24.0-5", "port-version": 2 },
    { "name": "atkmm", "version-string": "2.24.2", "port-version": 5 },
    { "name": "cairo", "version-string": "1.16.0", "port-version": 10 },
    { "name": "cairomm", "version-string": "1.15.3", "port-version": 6 },
    { "name": "gdk-pixbuf", "version-string": "2.36.9-5", "port-version": 7 },
    { "name": "glib", "version-string": "2.52.3", "port-version": 27 },
    { "name": "glibmm", "version-string": "2.52.1", "port-version": 14 },
    { "name": "gtk", "version-string": "3.22.19-4", "port-version": 0 },
    { "name": "gtkmm", "version-string": "3.22.2", "port-version": 5 },
    { "name": "harfbuzz", "version-string": "2.7.4", "port-version": 1 },
    { "name": "libsigcpp", "version-string": "2.10-3", "port-version": 2 },
    { "name": "pango", "version-string": "1.40.11", "port-version": 9 },
    { "name": "pangomm", "version-string": "2.40.1", "port-version": 4 },
    { "name": "pixman", "version-string": "0.38.4", "port-version": 2 },
    { "name": "tool-meson", "version-string": "0.56.02", "port-version": 0 }
  ]
```

**Comment by dg0yt** [comment]
In any case, upgrading gtk to version 4 broke port wxwidgets on linux.

**Comment by Neumann-A** [comment]
wxwidgets was put on skiplist by #11757 which was merged after #13100. In #13100 linux CI was green. So it probably had a cascade failure before that. I don't mind somebody adding a gtk3 port (using the native buildsystem of gtk and not the previous custom cmakelists.txt) since it is just like qt5 and qt6 and probably installs into separate folders for different versions. It just breaks the msbuild integration (which i don't really care about).
